db_main.go

```diff
@@ -100,6 +102,31 @@ func (db *DumbDB) Get(key []byte, bucket string) (ret_val []byte, err error) {
 	return
 }
 
+func (db *DumbDB) GetMultiple(keys [][]byte, bucket string) (ret_val [][]byte, err error) {
+
+	ret_val = make([][]byte, len(keys))
+	err = db.dbP.View(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket([]byte(bucket))
+		if bkt == nil {
+			db.err_log.Println("Bucket not created yet.")
+			return bolt.ErrBucketNotFound
+		}
+		for _,key := range keys{
+			ret_val = append(ret_val, bkt.Get(key))
```

Comment:
This is wrong. You are checking ret_val for nil. This will only work if the first value is nil. Once ret_val contains elements, you can append nil to it and ret_val will never become nil.

You need to check return value of bkt.Get and throw error if this is nil. Then append it to ret_val.